src/App.tsx

```diff
@@ -33,8 +34,22 @@ const App: React.FC = () => {
               <Switch>
                <Route exact path="/"
                   render={props => {
-                    return <Redirect to='/authenticated/myaccount' />
-                  }} />
+                    return <SynapseContextConsumer>
+                      {(ctx?: SynapseContextType) => {
+                        if (!ctx?.accessToken) {
+                          return <LoginPage returnToUrl={'/'} />
+                        } else {
+                          return <AppContextConsumer>
+                          {appContext => (
+                            <>
+                              {appContext?.redirectURL && <RedirectPage redirectURL={appContext.redirectURL}/>}
```

Comment:
Why not use <Redirect or   window.location.replace directly. Also does it need to be within a Consumer since we are supplying redirectURL directly?

Reply:
I thought <Redirect> only supported routes that react-router controlled, and I was thinking that you might prefer a component wrapper.  Removed RedirectPage to use replace() directly.
Using a Consumer because I think the redirectURL param might get lost in the Google SSO (or RAS SSO) login workflow, so I rely on the AppInitializer to pick up the redirectURL from either a query param or the local storage.